pkg/k8s/init.go

```diff
@@ -157,35 +155,8 @@ func WaitForNodeInformation(ctx context.Context, k8sGetter k8sGetter) error {
 
 		useNodeCIDR(n)
 
-		// Note: Node IPs are derived regardless of
-		// option.Config.EnableIPv4 and
-		// option.Config.EnableIPv6. This is done to enable
-		// underlay addressing to be different from overlay
-		// addressing, e.g. an IPv6 only PodCIDR running over
-		// IPv4 encapsulation.
-		if nodeIP4 != nil {
-			node.SetIPv4(nodeIP4)
-		}
-
-		if nodeIP6 != nil {
-			node.SetIPv6(nodeIP6)
-		}
-
-		node.SetLabels(n.Labels)
-
-		if option.Config.NodeEncryptionOptOutLabels.Matches(k8sLabels.Set(n.Labels)) {
-			log.WithField(logfields.Selector, option.Config.NodeEncryptionOptOutLabels).
-				Infof("Opting out from node-to-node encryption on this node as per '%s' label selector",
-					option.NodeEncryptionOptOutLabels)
-			node.SetOptOutNodeEncryption(true)
-		}
-
-		node.SetK8sExternalIPv4(n.GetExternalIP(false))
-		// Avoid removing the IPv4 if there is no IPv6.
-		if extIP6 := n.GetExternalIP(true); extIP6 != nil {
-			node.SetK8sExternalIPv6(extIP6)
-		}
-
+		// FIXME: This only works if we retrieved CiliumNode, but that
```

Comment:
Is there an issue for this?

Reply:
Oops, forgot follow through on this. Thanks.

Reply:
Double check ParseNode. Annotation should carry this and this is OK?